projects/storefrontlib/src/cms-components/checkout/guards/checkout-auth.guard.ts

```diff
@@ -1,29 +1,38 @@
 import { Injectable } from '@angular/core';
-import { CanActivate, Router } from '@angular/router';
+import { CanActivate } from '@angular/router';
 import {
   AuthRedirectService,
   AuthService,
+  CartService,
   RoutingService,
-  UserToken,
 } from '@spartacus/core';
-import { Observable } from 'rxjs';
+import { combineLatest, Observable } from 'rxjs';
 import { map } from 'rxjs/operators';
 
 @Injectable({
   providedIn: 'root',
 })
 export class CheckoutAuthGuard implements CanActivate {
   constructor(
-    protected routingService: RoutingService,
-    protected authService: AuthService,
```

Comment:
Not sure whether we need this. It seems we only need to check the cart's assigned user. If user's name is not "Anonymous", then it return true.

Reply:
I agree it works without it. I was thinking what if a customer doesn't need guest checkout and says well I'm going to modify the config for the cart to get cart and remove user. In this case checkout will be inaccessible (if the authservice is removed). What do you think?